src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -171,22 +233,27 @@ def get_matching_privacy_declarations(db: Session) -> Query:
             System.privacy_policy.label("system_privacy_policy"),
             System.vendor_id,
             PrivacyDeclaration.data_use,
-            PrivacyDeclaration.legal_basis_for_processing,
+            legal_basis_override_subquery.c.overridden_legal_basis_for_processing.label(  # pylint: disable=no-member
+                "legal_basis_for_processing"
+            ),
             PrivacyDeclaration.features,
             PrivacyDeclaration.retention_period,
+            PrivacyDeclaration.purpose,
+            PrivacyDeclaration.legal_basis_for_processing.label(
+                "original_legal_basis_for_processing"
+            ),
         )
         .outerjoin(PrivacyDeclaration, System.id == PrivacyDeclaration.system_id)
+        .outerjoin(
+            legal_basis_override_subquery,
+            legal_basis_override_subquery.c.id == PrivacyDeclaration.id,
+        )
```

Comment:
This is now the primary change of this PR . I join the base query to the legal basis override subquery, and then below filter on the legal basis override instead of the original legal basis.